core/trino-main/src/main/java/io/trino/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -1303,4 +1331,51 @@ public PlanNode getRoot()
             return mappings;
         }
     }
+
+    private static class DynamicFilterVisitor
+            extends SimplePlanRewriter<Void>
+    {
+        private final Metadata metadata;
+        private final Map<DynamicFilterId, DynamicFilterId> dynamicFilterIdMap;
+
+        private DynamicFilterVisitor(Metadata metadata, Map<DynamicFilterId, DynamicFilterId> dynamicFilterIdMap)
+        {
+            this.metadata = requireNonNull(metadata, "metadata is null");
+            this.dynamicFilterIdMap = requireNonNull(dynamicFilterIdMap, "dynamicFilterIdMap is null");
+        }
+
+        @Override
+        public PlanNode visitFilter(FilterNode node, RewriteContext<Void> context)
+        {
+            PlanNode rewrittenSource = context.rewrite(node.getSource());
+            Expression rewrittenPredicate = updateDynamicFilterIds(dynamicFilterIdMap, node.getPredicate());
+
+            if (rewrittenSource == node.getSource() && rewrittenPredicate == node.getPredicate()) {
+                return node;
+            }
+            return new FilterNode(
+                    node.getId(),
+                    rewrittenSource,
+                    rewrittenPredicate);
+        }
+
+        private Expression updateDynamicFilterIds(Map<DynamicFilterId, DynamicFilterId> dynamicFilterIdMap, Expression predicate)
+        {
+            List<Expression> conjuncts = extractConjuncts(predicate);
+            boolean[] updated = new boolean[1]; // using array as a mutable boolean to modify it inside the lambda
+            List<Expression> mapped = conjuncts.stream().map(conjunct -> getDescriptor(conjunct)
+                            .map(descriptor -> dynamicFilterIdMap.get(descriptor.getId()))
+                            .map(mappedId -> {
+                                updated[0] = true;
+                                return replaceDynamicFilterId((FunctionCall) conjunct, mappedId);
+                            })
+                            .orElse(conjunct))
+                    .collect(toImmutableList());
```

Comment:
Please move `.map(conjunct -> getDescriptor(conjunct)` to a new line. It will make the code more readable wrt which `map` applies to stream, and which applies to an Optional.